internal/model/todos.go

```diff
@@ -19,81 +19,119 @@ type TodoList struct {
 	Todos []*Todo `json:"todos"`
 }
 
-// GetTodosListByID returns slice of Todos based on the listID passed in
-func GetTodosListByID(listID string) (*TodoList, error) {
-	listStatement := `SELECT id, name FROM lists WHERE id = $1`
+// TodoListByID fetches a list and its todos
+func TodoListByID(id string) (*TodoList, error) {
 	list := &TodoList{}
 
-	err := db.QueryRow(listStatement, listID).Scan(&list.ID, &list.Name)
+	sql := "SELECT id, name FROM lists WHERE id = $1"
+	err := db.QueryRow(sql, id).Scan(&list.ID, &list.Name)
 	if err != nil {
 		return nil, err
 	}
 
-	sqlStatement := `SELECT id, list_id, todo, done FROM todos WHERE list_id = $1 ORDER BY created_at ASC;`
+	list.Todos, err = TodosByListID(id)
```

Comment:
Here we defer the loading of []*Todo (the todos) to a new func `TodosByListID`, let's us break things apart a bit and provides the extra functionality to our API for free.